arybutok/profiler-documentation-update

aprybutok opened the pull request:
Resolves #1 

Updates:

- Added container information to profiler documentation in source code
- Recompiled javadocs and jar to contain updated documentation
- Added link to documentation in READMEs using GitHub Pages

aprybutok committed:
Added information on container structure to growth, lysis, and parameter profiler documentation.

aprybutok committed:
Regenerated javadocs to reflect updated documentation.

aprybutok committed:
Rebuilt jar to contain updated documentation.

aprybutok committed:
Added link to GitHub Pages documentation for web browser viewing.

jessicasyu reviewed (commented):
-  For maintainability: generate the javadocs without the timestamp so in the future, if you need to make additional doc updates/changes, only the relevant javadocs will be marked as modified, rather than all 100+ files
- For readability: aim for 80 character (max 100 characters) per line and don't mix/match spaces and tabs. The really long lines in the javadoc (e.g. `LysisProfiler` line 70/71) are hard to read plus the mix of spaces/tabs in the files means that the formatting of the JSON in the output javadoc looks different
- I don't think the lines "The container is formatted as" is accurate. What container? It should probably be "The output JSON is formatted as"

aprybutok committed:
Fixed tabs/spaces in profiler documentation and regenerated javadocs and jar to reflect updates.

aprybutok commented:
@jessicasyu 

- Fixed spaces/tabs
- Fixed phrasing of "container" to "output JSON"
- Tried to fix the timestamp on the javadoc thing but couldn't figure out setting so tried a thing and wound up generating javadoc for only one file and then had to regenerate them all because in doing only the files I edited it _only_ showed those and not the other ones

But in the mean time these updates should be sufficient to resolve #1
